File: SampleProject/SampleProject/FirstViewController.m
//
//  FirstViewController.m
//  SampleProject
//
//  Created by Pollfish Inc. on 11/3/15.
//  Copyright © 2015 POLLFISH. All rights reserved.
//

#import "FirstViewController.h"

@interface FirstViewController ()

@end

@implementation FirstViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    if (@available(iOS 14, *)) {
        [self requestIDFAPermission];
    } else {
        [self initPollfish];
    }
}

- (void)requestIDFAPermission {
#if __has_include(<AppTrackingTransparency/AppTrackingTransparency.h>)
    [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
      dispatch_async(dispatch_get_main_queue(), ^{
          if (status == ATTrackingManagerAuthorizationStatusAuthorized) {
              [self initPollfish];
          } else {
              [self showNoPermissionAlert];
          }
      });
    }];
#endif
}

- (void)initPollfish {
    PollfishParams *pollfishParams = [[PollfishParams alloc] init:@"YOUR_API_KEY"];
    [pollfishParams indicatorPadding:10];
    [pollfishParams indicatorPosition:IndicatorPositionMiddleRight];
    [pollfishParams releaseMode:false];
    [pollfishParams offerwallMode:false];
    [pollfishParams rewardMode:false];
    [pollfishParams requestUUID:@"my_id"];
    
    [Pollfish initWith:pollfishParams delegate:self];
}

- (IBAction)showPollfish:(id)sender {
    
    NSLog(@"Pollfish: showPollfish");
    
    [Pollfish show];
}

- (IBAction)hidePollfish:(id)sender {
    NSLog(@"Pollfish: hidePollfish");
    
    [Pollfish hide];
}

- (void)pollfishSurveyReceivedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    int surveyPrice = [[surveyInfo cpa] intValue];
    int surveyIR = [[surveyInfo ir] intValue];
    int surveyLOI = [[surveyInfo loi] intValue];
    
    NSString *surveyClass = [surveyInfo surveyClass];
    
    NSString *rewardName = [surveyInfo rewardName];
    int rewardValue = [[surveyInfo rewardValue] intValue];
    
    NSLog(@"Pollfish: Survey Received - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue);
    
    _loggingLabel.text=@"Pollfish - Survey Received";
}

-(void)pollfishSurveyCompletedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    int surveyPrice = [[surveyInfo cpa] intValue];
    int surveyIR = [[surveyInfo ir] intValue];
    int surveyLOI = [[surveyInfo loi] intValue];
    
    NSString *surveyClass = [surveyInfo surveyClass];
    
    NSString *rewardName = [surveyInfo rewardName];
    int rewardValue = [[surveyInfo rewardValue] intValue];
    
    NSLog(@"Pollfish: Survey Completed - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue);
    
    _loggingLabel.text=@"Pollfish - Survey Completed";
}

- (void)pollfishOpened
{
    NSLog(@"Pollfish: Survey Opened");
    
    _loggingLabel.text=@"Pollfish - Survey Panel Opened";
}

- (void)pollfishClosed
{
    NSLog(@"Pollfish: Survey Closed");
    
    _loggingLabel.text=@"Pollfish - Survey Panel Closed";
}

- (void)pollfishNotAvailable
{
    NSLog(@"Pollfish: Survey Not Available");
    
    _loggingLabel.text=@"Pollfish - Survey Not Available";
}

- (void) pollfishUserRejectedSurvey
{
    NSLog(@"Pollfish: User Rejected Survey");
    
    _loggingLabel.text=@"Pollfish - User Rejected Survey";
}


- (void)pollfishUsernotEligible
{
    NSLog(@"Pollfish: User Not Eligible");
    
    _loggingLabel.text=@"Pollfish - User Not Eligible";
}

@end

File: SampleProjectSwift/SampleProjectSwift/FirstViewController.swift
//
//  FirstViewController.swift
//  SampleProjectSwift
//
//  Created by Pollfish Inc. on 11/4/15.
//  Copyright © 2015 POLLFISH. All rights reserved.
//

import UIKit

class FirstViewController: UIViewController {

    @IBOutlet weak var loggingLabel: UILabel!
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
       
        print("viewWillAppear()")
       
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishNotAvailable) , name:
            NSNotification.Name(rawValue: "PollfishSurveyNotAvailable"), object: nil)
       
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishReceived) , name:
            NSNotification.Name(rawValue: "PollfishSurveyReceived"), object: nil)

        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishOpened) , name:
            NSNotification.Name(rawValue: "PollfishOpened"), object: nil)
        
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishClosed) , name:
            NSNotification.Name(rawValue: "PollfishClosed"), object: nil)
        
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishUsernotEligible) , name:
            NSNotification.Name(rawValue: "PollfishUserNotEligible"), object: nil)
        
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.pollfishCompleted) , name:
            NSNotification.Name(rawValue: "PollfishSurveyCompleted"), object: nil)
        
        NotificationCenter.default.addObserver(self, selector:#selector(FirstViewController.rotateApp) , name:
          NSNotification.Name.UIDeviceOrientationDidChange, object: nil)
        
        Pollfish.initAtPosition( Int32(PollfishPosition.PollFishPositionMiddleLeft.rawValue), withPadding: 0, andDeveloperKey: "2ae349ab-30b8-4100-bc4d-b33b82e76519", andDebuggable: true, andCustomMode: false);
        
        loggingLabel.text="Logging area.."
    }
    
    func rotateApp(){
      loggingLabel.text="Logging area.."
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        print("viewDidLoad()")
    }
    
    func pollfishNotAvailable() {
        print("pollfishNotAvailable")
        
        loggingLabel.text="Pollfish - Survey Not Available"
    }
    
    func pollfishReceived() {
        print("pollfishReceived")
        
        loggingLabel.text="Pollfish - Survey Received"
    }
    
    func pollfishOpened() {
        print("pollfishOpened")
        
        loggingLabel.text="Pollfish - Pollfish Panel Opened"
    }
    
    func pollfishClosed() {
        print("pollfishClosed")

        loggingLabel.text="Pollfish - Pollfish Panel Closed"
    }
    
    func pollfishUsernotEligible() {
        print("pollfishUsernotEligible")
        
        loggingLabel.text="Pollfish - User Not Eligible"
    }
    
    func pollfishCompleted() {
        print("pollfishCompleted")
        
        loggingLabel.text="Pollfish - Pollfish Survey Completed"
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillDisappear(_ animated:Bool) {
        super.viewWillDisappear(animated)
        print("viewWillDisappear()")
        
        NotificationCenter.default.removeObserver(self)
    }
    
    
    @IBAction func showPollfish(_ sender: AnyObject) {
        print("showPollfish")
        
        Pollfish.show();
    }
    
    @IBAction func hidePollfish(_ sender: AnyObject) {
        print("hidePollfish")
        
        Pollfish.hide();
    }

}

